spring-integration-http/src/main/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandler.java

```diff
@@ -79,40 +52,12 @@
  * @author Gunnar Hillert
  * @author Artem Bilan
  * @author Wallace Wadge
+ * @author Shiliang Li
  * @since 2.0
  */
-public class HttpRequestExecutingMessageHandler extends AbstractReplyProducingMessageHandler {
-
-	private final Map<String, Expression> uriVariableExpressions = new HashMap<>();
-
+public class HttpRequestExecutingMessageHandler extends AbstractHttpRequestExecutingMessageHandler {
```

Comment:
M-m-m. Need to revise that doc.
You definitely shouldn't not `merge` for top level branch.
That is our responsibility for `upstream master`.

The top-level branch must be `rebase master` to be up-to-date.

You see those problems because your branch is far away from the `upstream master`.

Right now I can suggest you a simplest way if you can manage those conflicts.

Switch to the `master`, `pull upstream master` to have your local copy up-to-date.
Start a new top-level branch, e.g. `INT-4208b`.
Copy/paste your changes from here to there.
Push a new Pull Request for that new branch.
And close this one in favor of that.